helm/chart/router/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,16 @@
+{{- if .Values.serviceMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ .Release.Name }}
+spec:
+  endpoints:
+  - path: {{ .Values.router.configuration.telemetry.metrics.prometheus.path | default "/metrics" }}
```

Comment:
I think we need to be a bit more defensive here. We need to "chase down" the list of attributes to avoid errors. Something like:
```
  {{- if ((((.Values.router).configuration).telemetry).metrics).prometheus }}
```
is required to protect that setting.

Reply:
Should defaults for the entire configuration just be set in helm values? I think that would be a great step towards documenting the configuration.

Reply:
That's not the worst suggestion, it just doesn't seem to be very "helm typical". My concern would be having to keep the helm configuration aligned with the main router configuration. I can't decide if that would be harder or easier if we did what you suggest.

That's going to take some thinking through. Meanwhile, the defensive step here is required and we can remove it if we ever do go down the full-blown configuration setting route.

BTW: are you aware of `router --schema` ? I'm just wondering if you find that helpful or not? I find it reasonably helpful when working with router configuration.

Reply:
Point taken. I can see the benefits/drawbacks to either solution. Personally, I think it would be a benefit to have each configuration change essentially versioned with the chart so you know exactly what you're getting. It certainly adds maintenance overhead but as an end-user it would be really nice. I'm sure as helm chart testing in this repository develops, it could also be checked/validated in CI (using `chart-testing` or something) to catch issues.

I've used `router --schema` - the problem is that it's a bit difficult to parse because it's the entire/verbose schema definition rather than a flat set of k/v pairs. Maybe there's some `jq/yq` magic to be done to make it more readable, but I feel like it should be more transparent to end-users overall.